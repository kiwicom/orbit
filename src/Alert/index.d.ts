// @flow
// Type definitions for @kiwicom/orbit-components
// Project: http://github.com/kiwicom/orbit-components

import * as React from "react";

import * as Common from "../common/common";

type Type = "info" | "success" | "warning" | "critical";

declare module "@kiwicom/orbit-components/lib/Alert";

export interface Props extends Common.Global, Common.SpaceAfter {
  readonly type?: Type;
  readonly children?: React.ReactNode;
  readonly title?: Common.Translation;
  readonly icon?: boolean | React.ElementType;
  readonly closable?: boolean;
  readonly inlineActions?: React.ReactNode;
  readonly onClose?: Common.Callback;
}

declare const Alert: React.FunctionComponent<Props>;
export { Alert, Alert as default };
