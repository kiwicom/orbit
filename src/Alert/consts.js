// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
};

export const TOKENS = {
  backgroundAlert: "backgroundAlert",
  colorIconAlert: "colorIconAlert",
  colorTextAlert: "colorTextAlert",
  colorTextLinkAlertHover: "colorTextLinkAlertHover",
  colorTextLinkAlertFocus: "colorTextLinkAlertFocus",
  colorBorderAlert: "colorBorderAlert",
};

export const CLOSE_BUTTON_DATA_TEST = "AlertCloseButton";
