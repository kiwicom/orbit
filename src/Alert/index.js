// @flow
import * as React from "react";
import styled from "styled-components";
import defaultTokens from "@kiwicom/orbit-design-tokens";
import { withTheme } from "theming";

import {
  InformationCircle,
  Check,
  Alert as AlertTriangle,
  AlertCircle,
  CloseCircle,
} from "../icons";
import { Button } from "../index";

export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
};

type IconProps = {
  icon: React.Node,
  type: string,
};

const Icon = ({ icon, type }: IconProps) => {
  // Icon should be boolean and TRUE
  if (typeof icon === "boolean" && icon) {
    if (type === TYPE_OPTIONS.INFO) {
      return <InformationCircle />;
    }
    if (type === TYPE_OPTIONS.SUCCESS) {
      return <Check />;
    }
    if (type === TYPE_OPTIONS.WARNING) {
      return <AlertTriangle />;
    }
    if (type === TYPE_OPTIONS.CRITICAL) {
      return <AlertCircle />;
    }
  }
  return icon;
};

type Props = {
  type: $Values<typeof TYPE_OPTIONS>,
  title?: string,
  icon?: React.Element<any> | boolean,
  closable: boolean,
  onClose: () => void,
  children: React.Node,
  theme: typeof defaultTokens,
};

const StyledAlert = styled(({ theme, tokens, type, icon, ...props }) => <div {...props} />)`
  position: relative;
  display: flex;
  width: 100%;
  padding: ${({ theme, icon }) =>
    icon ? `${theme.paddingAlert} ${theme.paddingAlertWithIcon}` : theme.paddingAlert};
  padding-right: ${({ theme, closable }) => closable && theme.spaceXXLarge};
  border-radius: ${({ theme }) => theme.borderRadiusNormal};
  background: ${({ tokens, type }) => tokens.backgroundAlert[type]};
  color: ${({ tokens, type }) => tokens.colorTextAlert[type]};
  font-family: ${({ theme }) => theme.fontFamily};
  font-size: ${({ theme }) => theme.fontSizeTextNormal};
  box-sizing: border-box;
`;

const IconContainer = styled(({ theme, tokens, type, ...props }) => <div {...props} />)`
  flex-shrink: 0;
  margin-right: ${({ theme }) => theme.spaceSmall};
  color: ${({ tokens, type }) => tokens.colorIconAlert[type]};
`;

const Title = styled(({ theme, ...props }) => <div {...props} />)`
  display: flex;
  align-items: center;
  margin-bottom: ${({ theme }) => theme.spaceXSmall};
  font-weight: ${({ theme }) => theme.fontWeightBold};
  line-height: ${({ theme }) => theme.lineHeightHeading};
  min-height: ${({ theme }) => theme.heightIconMedium};
`;

const Content = styled(({ theme, title, ...props }) => <div {...props} />)`
  display: flex;
  align-items: center;
  min-height: ${({ theme, title }) => !title && theme.heightIconMedium};
  margin-bottom: ${({ theme, title }) => title && theme.spaceXXSmall};
`;

const CloseContainer = styled(({ theme, ...props }) => <div {...props} />)`
  position: absolute;
  top: ${({ theme }) => theme.spaceXXSmall};
  right: ${({ theme }) => theme.spaceXXSmall};
`;

const Alert = (props: Props) => {
  const { type, title, theme, closable, icon, onClose, children } = props;
  const tokens = {
    colorIconAlert: {
      [TYPE_OPTIONS.INFO]: theme.colorAlertIconInfo,
      [TYPE_OPTIONS.SUCCESS]: theme.colorAlertIconSuccess,
      [TYPE_OPTIONS.WARNING]: theme.colorAlertIconWarning,
      [TYPE_OPTIONS.CRITICAL]: theme.colorAlertIconCritical,
    },
    backgroundAlert: {
      [TYPE_OPTIONS.INFO]: theme.backgroundAlertInfo,
      [TYPE_OPTIONS.SUCCESS]: theme.backgroundAlertSuccess,
      [TYPE_OPTIONS.WARNING]: theme.backgroundAlertWarning,
      [TYPE_OPTIONS.CRITICAL]: theme.backgroundAlertCritical,
    },
    colorTextAlert: {
      [TYPE_OPTIONS.INFO]: theme.colorTextAlertInfo,
      [TYPE_OPTIONS.SUCCESS]: theme.colorTextAlertSuccess,
      [TYPE_OPTIONS.WARNING]: theme.colorTextAlertWarning,
      [TYPE_OPTIONS.CRITICAL]: theme.colorTextAlertCritical,
    },
  };
  return (
    <StyledAlert tokens={tokens} {...props}>
      {icon && (
        <IconContainer theme={theme} tokens={tokens} type={type}>
          <Icon type={type} icon={icon} />
        </IconContainer>
      )}
      <div>
        {title && <Title theme={theme}>{title}</Title>}
        <Content theme={theme} title={title}>
          {children}
        </Content>
      </div>
      {closable && (
        <CloseContainer theme={theme}>
          <Button
            onClick={onClose}
            type={type}
            variation="link"
            size="small"
            Icon={CloseCircle}
            onlyIcon
          />
        </CloseContainer>
      )}
    </StyledAlert>
  );
};

const AlertWithTheme = withTheme(Alert);
AlertWithTheme.displayName = "Alert";
AlertWithTheme.defaultProps = {
  type: "info",
  closable: false,
  onClose: () => {},
};

export default AlertWithTheme;
