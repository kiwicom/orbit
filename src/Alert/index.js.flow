// @flow
/*
  DOCUMENTATION: https://orbit.kiwi/components/alert-message/
*/
import type { spaceAfter } from "../common/getSpacingToken/index";
import type { Globals, Translation } from "../common/common.js.flow";

type Type = "info" | "success" | "warning" | "critical";

export type Props = {|
  +type?: Type,
  +children?: React$Node,
  +title?: Translation,
  +icon?: React$Element<any> | boolean,
  +closable?: boolean,
  +inlineActions?: React$Node,
  +onClose?: () => void | Promise<any>,
  ...Globals,
  ...spaceAfter,
|};

declare export default React$ComponentType<Props>;
