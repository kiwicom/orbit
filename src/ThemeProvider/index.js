// @flow
import React from "react";
import { ThemeProvider as StyledThemeProvider } from "styled-components";

import Dictionary from "../Dictionary";

import type { Props } from "./index";

const ThemeProvider = ({ theme, dictionary, children }: Props) => (
  <StyledThemeProvider theme={theme}>
    {dictionary ? (
      <Dictionary values={dictionary}>{React.Children.only(children)}</Dictionary>
    ) : (
      React.Children.only(children)
    )}
  </StyledThemeProvider>
);

export default ThemeProvider;
