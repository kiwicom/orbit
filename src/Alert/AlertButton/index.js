// @flow
import * as React from "react";

import { TYPE_OPTIONS } from "./consts";
import ButtonPrimitive from "../../primitives/ButtonPrimitive";
import getIconContainer from "../../primitives/ButtonPrimitive/common/getIconContainer";
import getCommonProps from "../../primitives/ButtonPrimitive/common/getCommonProps";
import useTheme from "../../hooks/useTheme";
import getAlertButtonStyles from "./helpers/getAlertButtonStyles";
import getAlertButtonIconForeground from "./helpers/getAlertButtonIconForeground";

import type { Props } from "./index";

const AlertButton = React.forwardRef<Props, HTMLButtonElement>(
  ({ type = TYPE_OPTIONS.INFO, disabled = false, ...props }, ref) => {
    const theme = useTheme();
    const propsWithTheme = { theme, ...props };
    const commonProps = getCommonProps(propsWithTheme);
    const buttonStyles = getAlertButtonStyles({ type, theme, disabled });
    const icons = getIconContainer({
      ...propsWithTheme,
      iconForeground: getAlertButtonIconForeground({ type, theme }),
    });
    return (
      <ButtonPrimitive
        ref={ref}
        disabled={disabled}
        {...props}
        {...buttonStyles}
        {...commonProps}
        {...icons}
      />
    );
  },
);

AlertButton.displayName = "AlertButton";

export default AlertButton;
