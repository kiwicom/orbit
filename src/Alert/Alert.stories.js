// @flow
import * as React from "react";
import { storiesOf, setAddon } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import styles from "@sambego/storybook-styles";
import chaptersAddon from "react-storybook-addon-chapters";
import { withKnobs, text, boolean, select } from "@storybook/addon-knobs/react";

import ThemeProvider from "../Theming/ThemeProvider";
import * as Icons from "../icons";

import Alert, { TYPE_OPTIONS } from "./index";

setAddon(chaptersAddon);

const getIcons = defaultIcon => select("Icon", [undefined, ...Object.keys(Icons)], defaultIcon);
const getIcon = source => Icons[source];
const renderTitle = title => (title === "" ? undefined : title);

const options = {
  showSource: true,
  allowSourceToggling: false,
  showPropTables: false,
  allowPropTablesToggling: false,
};

storiesOf("Alert", module)
  .addDecorator(withKnobs)
  .addDecorator(
    styles({
      padding: "20px",
    }),
  )
  .addWithChapters("Default", () => {
    const message = "The quick, brown fox jumps over a lazy dog.";
    return {
      info:
        "This is the default configuration of this component. Visit Orbit.Kiwi for more detailed guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => (
                <ThemeProvider>
                  <Alert>{message}</Alert>
                </ThemeProvider>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Info alert", () => {
    const title = text("Title");
    const message = text("Message", "The quick, brown fox jumps over a lazy dog.");
    return {
      info:
        "Use when you need to inform users about something that is happening in their booking or a trip. If the issue is potentially dangerous, consider using warning alert. Keep in mind that warning alert can stress users more than the informational one. Visit Orbit.Kiwi for more detailed guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => (
                <ThemeProvider>
                  <Alert title={renderTitle(title)} icon>
                    {message}
                  </Alert>
                </ThemeProvider>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Success alert", () => {
    const title = text("Title");
    const message = text("Message", "The quick, brown fox jumps over a lazy dog.");
    return {
      info:
        "Use when a user just performed some action and we need to tell them that action was successful. This button is usually used without an action button.\n" +
        "Avoid using success banner if there is any follow-up action, for example, in cases where we need to confirm something to users by e-mail later. It's recommended to use informational alert instead. Visit Orbit.Kiwi for more detailed guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => (
                <ThemeProvider>
                  <Alert type="success" title={renderTitle(title)} icon>
                    {message}
                  </Alert>
                </ThemeProvider>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Warning alert", () => {
    const title = text("Title");
    const message = text("Message", "The quick, brown fox jumps over a lazy dog.");
    return {
      info:
        "Use in cases when you need to inform users about a potentially dangerous situation in their trip and it requires some action from them. However, if the issue requires immediate attention, use critical alert instead. Visit Orbit.Kiwi for more detailed guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => (
                <ThemeProvider>
                  <Alert type="warning" title={renderTitle(title)} icon>
                    {message}
                  </Alert>
                </ThemeProvider>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Critical alert", () => {
    const title = text("Title");
    const message = text("Message", "The quick, brown fox jumps over a lazy dog.");
    return {
      info:
        "Use when something is blocking users from continuing or when some issue needs to be resolved immediately. A critical alert should provide some form of solution for their problem. If something is important for users to solve as soon as possible, automatic open of a modal window is worthy of considering. Visit Orbit.Kiwi for more detailed guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => (
                <ThemeProvider>
                  <Alert type="critical" title={renderTitle(title)} icon>
                    {message}
                  </Alert>
                </ThemeProvider>
              ),
              options,
            },
          ],
        },
      ],
    };
  })
  .addWithChapters("Playground", () => {
    const type = select("Type", Object.values(TYPE_OPTIONS), "info");
    const title = text("Title", "You can change the title by changing the Title knob");
    const message = text("Message", "Also you can change the message by changing the Message knob");
    const closable = boolean("Closable", false);
    const Icon = getIcon(getIcons("Airplane"));
    return {
      info:
        "You can try all possible configurations of this component. However, check Orbit.Kiwi for more detailed design guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => (
                <ThemeProvider>
                  <Alert
                    type={type}
                    icon={Icon && <Icon />}
                    title={renderTitle(title)}
                    closable={closable}
                    onClose={action("Close")}
                  >
                    {message}
                  </Alert>
                </ThemeProvider>
              ),
              options,
            },
          ],
        },
      ],
    };
  });
