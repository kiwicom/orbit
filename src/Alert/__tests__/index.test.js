// @flow
import * as React from "react";
import { shallow, mount } from "enzyme";

import ThemeProvider from "../../Theming/ThemeProvider";
import Alert from "../index";

describe("Alert", () => {
  it("should contain children", () => {
    const message = "Alert message";
    const component = mount(
      <ThemeProvider>
        <Alert>{message}</Alert>
      </ThemeProvider>,
    );
    expect(
      component
        .find("Alert__Content")
        .children()
        .exists(),
    ).toBe(true);
  });
  // it("should be closable", () => {
  //   const onClose = jest.fn();
  //   const message = "Alert message";
  //   const component = mount(
  //     <ThemeProvider>
  //       <Alert onClose={onClose} closable>
  //         {message}
  //       </Alert>
  //     </ThemeProvider>,
  //   );
  //   const alert = component.find("Alert__CloseContainer");
  //   alert.simulate("click");
  //   expect(onClose).toHaveBeenCalled();
  // });
  it("should match snapshot", () => {
    const message = "Alert message";
    const component = shallow(
      <ThemeProvider>
        <Alert>{message}</Alert>
      </ThemeProvider>,
    );
    expect(component).toMatchSnapshot();
  });
});
