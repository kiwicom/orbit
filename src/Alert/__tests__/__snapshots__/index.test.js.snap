// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundAlertCritical": "#fae8e8",
      "backgroundAlertInfo": "#e0f6ff",
      "backgroundAlertSuccess": "#e7f3e8",
      "backgroundAlertWarning": "#fcf1cd",
      "backgroundBody": "#f5f7f9",
      "backgroundButtonBordered": "transparent",
      "backgroundButtonBorderedActive": "#fff",
      "backgroundButtonBorderedHover": "#f5f7f9",
      "backgroundButtonCritical": "#d21c1c",
      "backgroundButtonCriticalActive": "#b21717",
      "backgroundButtonCriticalHover": "#bd1919",
      "backgroundButtonCriticalLink": "transparent",
      "backgroundButtonCriticalLinkActive": "#f1c0c0",
      "backgroundButtonCriticalLinkHover": "#f5d4d4",
      "backgroundButtonFacebook": "#3b5998",
      "backgroundButtonFacebookActive": "#354f88",
      "backgroundButtonFacebookHover": "#385490",
      "backgroundButtonFacebookLink": "#3b5998",
      "backgroundButtonFacebookLinkActive": "#354f88",
      "backgroundButtonFacebookLinkHover": "#385490",
      "backgroundButtonGoogle": "#f5f7f9",
      "backgroundButtonGoogleActive": "#d6dee6",
      "backgroundButtonGoogleHover": "#e5eaef",
      "backgroundButtonGoogleLink": "transparent",
      "backgroundButtonGoogleLinkActive": "#d6dee6",
      "backgroundButtonGoogleLinkHover": "#e5eaef",
      "backgroundButtonInfo": "#0176D2",
      "backgroundButtonInfoActive": "#0064b2",
      "backgroundButtonInfoHover": "#006abd",
      "backgroundButtonInfoLink": "transparent",
      "backgroundButtonInfoLinkActive": "#b0e8fe",
      "backgroundButtonInfoLinkHover": "#c8effe",
      "backgroundButtonPrimary": "#00a991",
      "backgroundButtonPrimaryActive": "#008f7b",
      "backgroundButtonPrimaryHover": "#009882",
      "backgroundButtonPrimaryLink": "transparent",
      "backgroundButtonPrimaryLinkActive": "#73cac0",
      "backgroundButtonPrimaryLinkHover": "#80cfc6",
      "backgroundButtonSecondary": "#e8edf1",
      "backgroundButtonSecondaryActive": "#cad5df",
      "backgroundButtonSecondaryHover": "#d9e1e8",
      "backgroundButtonSecondaryLink": "transparent",
      "backgroundButtonSecondaryLinkActive": "#d6dee6",
      "backgroundButtonSecondaryLinkHover": "#e5eaef",
      "backgroundButtonSuccess": "#46B655",
      "backgroundButtonSuccessActive": "#3fa34c",
      "backgroundButtonSuccessHover": "#42ac50",
      "backgroundButtonSuccessLink": "transparent",
      "backgroundButtonSuccessLinkActive": "#c7e3c9",
      "backgroundButtonSuccessLinkHover": "#d7ebd8",
      "backgroundButtonWarning": "#f9971e",
      "backgroundButtonWarningActive": "#e68206",
      "backgroundButtonWarningHover": "#f48a06",
      "backgroundButtonWarningLink": "transparent",
      "backgroundButtonWarningLinkActive": "#f9e4a1",
      "backgroundButtonWarningLinkHover": "#faeab7",
      "backgroundCard": "#fff",
      "backgroundCarrierLogo": "#fff",
      "backgroundInput": "#fff",
      "backgroundInputDisabled": "#e8edf1",
      "backgroundModal": "#fff",
      "borderColorButtonCriticalBordered": "#d21c1c",
      "borderColorButtonCriticalBorderedActive": "#b21717",
      "borderColorButtonCriticalBorderedHover": "#bd1919",
      "borderColorButtonFacebookBordered": "#3b5998",
      "borderColorButtonFacebookBorderedActive": "#354f88",
      "borderColorButtonFacebookBorderedHover": "#385490",
      "borderColorButtonGoogleBordered": "#46515e",
      "borderColorButtonGoogleBorderedActive": "#38404b",
      "borderColorButtonGoogleBorderedHover": "#3f4854",
      "borderColorButtonInfoBordered": "#0176D2",
      "borderColorButtonInfoBorderedActive": "#0064b2",
      "borderColorButtonInfoBorderedHover": "#006abd",
      "borderColorButtonPrimaryBordered": "#00a991",
      "borderColorButtonPrimaryBorderedActive": "#008f7b",
      "borderColorButtonPrimaryBorderedHover": "#009882",
      "borderColorButtonSecondaryBordered": "#46515e",
      "borderColorButtonSecondaryBorderedActive": "#38404b",
      "borderColorButtonSecondaryBorderedHover": "#3f4854",
      "borderColorButtonSuccessBordered": "#46B655",
      "borderColorButtonSuccessBorderedActive": "#3fa34c",
      "borderColorButtonSuccessBorderedHover": "#42ac50",
      "borderColorButtonWarningBordered": "#f9971e",
      "borderColorButtonWarningBorderedActive": "#e68206",
      "borderColorButtonWarningBorderedHover": "#f48a06",
      "borderColorCard": "#e8edf1",
      "borderColorInput": "#bac7d5",
      "borderColorInputActive": "#94a8be",
      "borderColorInputError": "#d21c1c",
      "borderColorInputErrorFocus": "#d21c1c",
      "borderColorInputFocus": "#0176D2",
      "borderColorInputHover": "#a6b6c8",
      "borderColorModal": "#e8edf1",
      "borderColorTableCell": "#bac7d5",
      "borderRadiusCircle": "50%",
      "borderRadiusLarge": "6px",
      "borderRadiusNormal": "3px",
      "borderRadiusSmall": "2px",
      "borderStyleCard": "solid",
      "borderStyleInput": "solid",
      "borderWidthCard": "1px",
      "borderWidthInput": "1px",
      "boxShadowInputErrorFocus": "0px 0px 3px 1px rgba(210,28,28,0.3)",
      "boxShadowInputFocus": "0px 0px 3px 1px rgba(1,118,210,0.3)",
      "colorAlertIconCritical": "#d21c1c",
      "colorAlertIconInfo": "#0176D2",
      "colorAlertIconSuccess": "#46B655",
      "colorAlertIconWarning": "#f9971e",
      "colorHeading": "#171b1e",
      "colorIconAttention": "#171b1e",
      "colorIconCheckbox": "#00a991",
      "colorIconCheckboxDisabled": "#bac7d5",
      "colorIconInput": "#bac7d5",
      "colorIconPrimary": "#46515e",
      "colorIconRadioButton": "#00a991",
      "colorIconRadioButtonDisabled": "#bac7d5",
      "colorIconSecondary": "#7f91a8",
      "colorIconTerciary": "#bac7d5",
      "colorLabelForm": "#46515e",
      "colorLabelFormFilled": "#7f91a8",
      "colorLinkPrimary": "#00a991",
      "colorLinkPrimaryHover": "#00a991",
      "colorLinkSecondary": "#171b1e",
      "colorPlaceholderInput": "#bac7d5",
      "colorTextAlertCritical": "#650808",
      "colorTextAlertInfo": "#07405c",
      "colorTextAlertSuccess": "#065d12",
      "colorTextAlertWarning": "#a93610",
      "colorTextAttention": "#171b1e",
      "colorTextButtonCritical": "#fff",
      "colorTextButtonCriticalActive": "#fff",
      "colorTextButtonCriticalBordered": "#d21c1c",
      "colorTextButtonCriticalBorderedActive": "#b21717",
      "colorTextButtonCriticalBorderedHover": "#bd1919",
      "colorTextButtonCriticalHover": "#fff",
      "colorTextButtonCriticalLink": "#d21c1c",
      "colorTextButtonCriticalLinkActive": "#b21717",
      "colorTextButtonCriticalLinkHover": "#bd1919",
      "colorTextButtonFacebook": "#fff",
      "colorTextButtonFacebookActive": "#fff",
      "colorTextButtonFacebookBordered": "#3b5998",
      "colorTextButtonFacebookBorderedActive": "#354f88",
      "colorTextButtonFacebookBorderedHover": "#385490",
      "colorTextButtonFacebookHover": "#fff",
      "colorTextButtonFacebookLink": "#fff",
      "colorTextButtonFacebookLinkActive": "#fff",
      "colorTextButtonFacebookLinkHover": "#fff",
      "colorTextButtonFilled": "#fff",
      "colorTextButtonFilledActive": "#fff",
      "colorTextButtonFilledHover": "#fff",
      "colorTextButtonGoogle": "#46515e",
      "colorTextButtonGoogleActive": "#38404b",
      "colorTextButtonGoogleBordered": "#46515e",
      "colorTextButtonGoogleBorderedActive": "#38404b",
      "colorTextButtonGoogleBorderedHover": "#3f4854",
      "colorTextButtonGoogleHover": "#3f4854",
      "colorTextButtonGoogleLink": "#46515e",
      "colorTextButtonGoogleLinkActive": "#38404b",
      "colorTextButtonGoogleLinkHover": "#3f4854",
      "colorTextButtonInfo": "#fff",
      "colorTextButtonInfoActive": "#fff",
      "colorTextButtonInfoBordered": "#0176D2",
      "colorTextButtonInfoBorderedActive": "#0064b2",
      "colorTextButtonInfoBorderedHover": "#006abd",
      "colorTextButtonInfoHover": "#fff",
      "colorTextButtonInfoLink": "#0176D2",
      "colorTextButtonInfoLinkActive": "#0064b2",
      "colorTextButtonInfoLinkHover": "#006abd",
      "colorTextButtonPrimary": "#fff",
      "colorTextButtonPrimaryActive": "#fff",
      "colorTextButtonPrimaryBordered": "#00a991",
      "colorTextButtonPrimaryBorderedActive": "#008f7b",
      "colorTextButtonPrimaryBorderedHover": "#009882",
      "colorTextButtonPrimaryHover": "#fff",
      "colorTextButtonPrimaryLink": "#00a991",
      "colorTextButtonPrimaryLinkActive": "#008f7b",
      "colorTextButtonPrimaryLinkHover": "#009882",
      "colorTextButtonSecondary": "#46515e",
      "colorTextButtonSecondaryActive": "#38404b",
      "colorTextButtonSecondaryBordered": "#46515e",
      "colorTextButtonSecondaryBorderedActive": "#38404b",
      "colorTextButtonSecondaryBorderedHover": "#3f4854",
      "colorTextButtonSecondaryHover": "#3f4854",
      "colorTextButtonSecondaryLink": "#46515e",
      "colorTextButtonSecondaryLinkActive": "#38404b",
      "colorTextButtonSecondaryLinkHover": "#3f4854",
      "colorTextButtonSuccess": "#fff",
      "colorTextButtonSuccessActive": "#fff",
      "colorTextButtonSuccessBordered": "#46B655",
      "colorTextButtonSuccessBorderedActive": "#3fa34c",
      "colorTextButtonSuccessBorderedHover": "#42ac50",
      "colorTextButtonSuccessHover": "#fff",
      "colorTextButtonSuccessLink": "#46B655",
      "colorTextButtonSuccessLinkActive": "#3fa34c",
      "colorTextButtonSuccessLinkHover": "#42ac50",
      "colorTextButtonWarning": "#fff",
      "colorTextButtonWarningActive": "#fff",
      "colorTextButtonWarningBordered": "#f9971e",
      "colorTextButtonWarningBorderedActive": "#e68206",
      "colorTextButtonWarningBorderedHover": "#f48a06",
      "colorTextButtonWarningHover": "#fff",
      "colorTextButtonWarningLink": "#f9971e",
      "colorTextButtonWarningLinkActive": "#e68206",
      "colorTextButtonWarningLinkHover": "#f48a06",
      "colorTextError": "#d21c1c",
      "colorTextInfo": "#0176D2",
      "colorTextInput": "#46515e",
      "colorTextInputDisabled": "#bac7d5",
      "colorTextPrimary": "#46515e",
      "colorTextSecondary": "#7f91a8",
      "durationFast": "0.2s",
      "durationNormal": "0.3s",
      "durationSlow": "0.4s",
      "fontFamily": "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
      "fontSizeButtonLarge": "16px",
      "fontSizeButtonNormal": "14px",
      "fontSizeButtonSmall": "12px",
      "fontSizeHeadingDisplay": "40px",
      "fontSizeHeadingLarge": "28px",
      "fontSizeHeadingMedium": "22px",
      "fontSizeHeadingSmall": "16px",
      "fontSizeHeadingTitle1": "28px",
      "fontSizeHeadingTitle2": "22px",
      "fontSizeHeadingTitle3": "16px",
      "fontSizeInputNormal": "16px",
      "fontSizeInputSmall": "14px",
      "fontSizeLabelForm": "14px",
      "fontSizeMessage": "14px",
      "fontSizeMessageForm": "12px",
      "fontSizeTextLarge": "16px",
      "fontSizeTextNormal": "14px",
      "fontSizeTextSmall": "12px",
      "fontWeightBold": "700",
      "fontWeightHeadingDisplay": "700",
      "fontWeightHeadingLevel1": "700",
      "fontWeightHeadingLevel2": "500",
      "fontWeightHeadingLevel3": "500",
      "fontWeightLinks": "500",
      "fontWeightMedium": "500",
      "fontWeightNormal": "400",
      "heightButtonLarge": "52px",
      "heightButtonNormal": "44px",
      "heightButtonSmall": "32px",
      "heightCheckbox": "20px",
      "heightIconLarge": "32px",
      "heightIconMedium": "24px",
      "heightIconSmall": "16px",
      "heightInputLarge": "52px",
      "heightInputNormal": "44px",
      "heightInputSmall": "32px",
      "heightRadioButton": "20px",
      "lineHeightHeading": "1.2",
      "lineHeightText": "1.4",
      "marginButtonIconLarge": "16px",
      "marginButtonIconNormal": "12px",
      "marginButtonIconSmall": "8px",
      "modifierDarkenActive": 0.1,
      "modifierDarkenButtonLink": 0.05,
      "modifierDarkenHover": 0.05,
      "modifierScaleButtonActive": 0.9,
      "opacityButtonDisabled": "0.3",
      "opacityCheckboxDisabled": "0.5",
      "opacityOverlay": "0.8",
      "opacityRadioButtonDisabled": "0.5",
      "paddingAlert": "16px",
      "paddingAlertWithIcon": "12px",
      "paddingButtonLarge": "28px",
      "paddingButtonLargeWithIcon": "12px",
      "paddingButtonNormal": "16px",
      "paddingButtonNormalWithIcon": "8px",
      "paddingButtonSmall": "12px",
      "paddingButtonSmallWithIcon": "8px",
      "paddingInputNormal": "16px",
      "paddingInputSmall": "12px",
      "paletteBlueDark": "#07405c",
      "paletteBlueLight": "#e0f6ff",
      "paletteBlueLightActive": "#b0e8fe",
      "paletteBlueLightHover": "#c8effe",
      "paletteBlueNormal": "#0176D2",
      "paletteBlueNormalActive": "#0064b2",
      "paletteBlueNormalHover": "#006abd",
      "paletteBrandDark": "#00826f",
      "paletteBrandLight": "#9cdad3",
      "paletteBrandLightActive": "#73cac0",
      "paletteBrandLightHover": "#80cfc6",
      "paletteBrandNormal": "#00a991",
      "paletteBrandNormalActive": "#008f7b",
      "paletteBrandNormalHover": "#009882",
      "paletteCloudLight": "#f5f7f9",
      "paletteCloudLightActive": "#d6dee6",
      "paletteCloudLightHover": "#e5eaef",
      "paletteCloudNormal": "#e8edf1",
      "paletteCloudNormalActive": "#cad5df",
      "paletteCloudNormalHover": "#d9e1e8",
      "paletteGreenDark": "#065d12",
      "paletteGreenLight": "#e7f3e8",
      "paletteGreenLightActive": "#c7e3c9",
      "paletteGreenLightHover": "#d7ebd8",
      "paletteGreenNormal": "#46B655",
      "paletteGreenNormalActive": "#3fa34c",
      "paletteGreenNormalHover": "#42ac50",
      "paletteInkDark": "#171b1e",
      "paletteInkLight": "#7f91a8",
      "paletteInkLightActive": "#5f738c",
      "paletteInkLightHover": "#6d819c",
      "paletteInkLighter": "#bac7d5",
      "paletteInkLighterActive": "#94a8be",
      "paletteInkLighterHover": "#a6b6c8",
      "paletteInkNormal": "#46515e",
      "paletteInkNormalActive": "#38404b",
      "paletteInkNormalHover": "#3f4854",
      "paletteOrangeDark": "#a93610",
      "paletteOrangeLight": "#fcf1cd",
      "paletteOrangeLightActive": "#f9e4a1",
      "paletteOrangeLightHover": "#faeab7",
      "paletteOrangeNormal": "#f9971e",
      "paletteOrangeNormalActive": "#e68206",
      "paletteOrangeNormalHover": "#f48a06",
      "paletteRedDark": "#650808",
      "paletteRedLight": "#fae8e8",
      "paletteRedLightActive": "#f1c0c0",
      "paletteRedLightHover": "#f5d4d4",
      "paletteRedNormal": "#d21c1c",
      "paletteRedNormalActive": "#b21717",
      "paletteRedNormalHover": "#bd1919",
      "paletteSocialFacebook": "#3b5998",
      "paletteSocialFacebookActive": "#354f88",
      "paletteSocialFacebookHover": "#385490",
      "paletteWhite": "#fff",
      "paletteWhiteActive": "#e5eaef",
      "paletteWhiteHover": "#f5f7f9",
      "spaceLarge": "24px",
      "spaceMedium": "16px",
      "spaceSmall": "12px",
      "spaceXLarge": "32px",
      "spaceXSmall": "8px",
      "spaceXXLarge": "40px",
      "spaceXXSmall": "4px",
      "spaceXXXLarge": "52px",
      "textDecorationLinkPrimary": "none",
      "textDecorationLinkPrimaryHover": "underline",
      "textDecorationLinkSecondary": "underline",
      "textDecorationLinkSecondaryHover": "underline",
      "widthCheckbox": "20px",
      "widthIconLarge": "32px",
      "widthIconMedium": "24px",
      "widthIconSmall": "16px",
      "widthRadioButton": "20px",
      "zIndexDefault": "1",
      "zIndexModal": "825",
      "zIndexModalOverlay": "800",
      "zIndexOnTheTop": "900",
      "zIndexSticky": "100",
    }
  }
>
  <Alert
    closable={false}
    onClose={[Function]}
    type="info"
  >
    Alert message
  </Alert>
</ThemeProvider>
`;
